### PR TITLE
VS-99: Optimize MQL Generator to avoid having 1 using statement for each Bson Type

### PR DESCRIPTION
**Pull request by Ravi-Raghavan**
VS-99: Optimize MQL Generator to avoid having 1 using statement for each Bson Type

### CODE REVIEW

File: src/MongoDB.Analyzer.Helpers/Linq/MqlGenerator.cs [hunk 1]
```diff
@@ -14,18 +14,18 @@
...
 
 namespace MongoDB.Analyzer.Helpers.Linq
 {
     public static class MqlGenerator
     {
 #pragma warning disable CS0169 // The field is never used
 #pragma warning disable IDE0051
-        private static readonly BsonTypeCustom123 s_dummyRef1;
-        private static readonly BsonTimeSpanCustom123 s_dummyRef2;
+        private static readonly BsonType s_dummyRef1;
```
Comment (BorisDog) [inline src/MongoDB.Analyzer.Helpers/Linq/MqlGenerator.cs:27]: Update the  comment please to "// These fields are never used, they are needed to ensure that the relevant usings are not accidently removed"

File: src/MongoDB.Analyzer/Core/TypesProcessor.cs [hunk 2]
```diff
@@ -73,9 +64,9 @@ public string ProcessTypeSymbol(ITypeSymbol typeSymbol)
         }
 
         var fullTypeName = GetFullName(typeSymbol);
-        if (s_knownBsonTypes.TryGetValue(fullTypeName, out var knowTypeName))
+        if (typeSymbol.IsSupportedBsonType() || typeSymbol.IsSupportedBsonSerializationOption())
```
Comment (BorisDog) [inline src/MongoDB.Analyzer/Core/TypesProcessor.cs:67]: These methods perform `ToDisplayString` twice, while we already have `fullTypeName`.
Lets minimize the lookups and imbed `IsSupportedBsonSerializationOption` into `IsSupportedBsonType` (`startsWith(NamespaceMongoDBBson)`), and receive optional parameter `fullTypeName`.

**Review by BorisDog — CHANGES_REQUESTED** [review]

File: src/MongoDB.Analyzer.Helpers/Linq/MqlGenerator.cs (hunk 1 above)
Comment (Ravi-Raghavan) [reply src/MongoDB.Analyzer.Helpers/Linq/MqlGenerator.cs]: Done

File: src/MongoDB.Analyzer/Core/TypesProcessor.cs (hunk 2 above)
Comment (Ravi-Raghavan) [reply src/MongoDB.Analyzer/Core/TypesProcessor.cs]: Done

**Review by BorisDog — APPROVED** [review]
LGTM (notice the small comment)

File: src/MongoDB.Analyzer/Core/Utilities/SymbolExtensions.cs [hunk 3]
```diff
@@ -50,6 +44,26 @@ internal static class SymbolExtensions
...
 
+    private static readonly HashSet<string> s_supportedBsonSerializationOptions = new()
+    {
+        "MongoDB.Bson.Serialization.Options.TimeSpanUnits"
+    };
+
+    private static readonly HashSet<string> s_supportedBsonTypes = new()
+    {
+        "MongoDB.Bson.BsonDocument",
+        "MongoDB.Bson.BsonValue",
```
[the remark this answers is not in the document]
Comment (Ravi-Raghavan) [reply src/MongoDB.Analyzer/Core/Utilities/SymbolExtensions.cs]: Done

**Pull request merged**